library/Backends/llvm/src/Juvix/Backends/LLVM/Codegen/Types/Shared.hs

```diff
@@ -0,0 +1,43 @@
+-- | Shared between Types and Sum
+module Juvix.Backends.LLVM.Codegen.Types.Shared
+  ( SymbolTable,
+    TypeTable,
+    SumInfo (..),
+    VariantToType,
+    Names,
+    uniqueName,
+  )
+where
+
+import Juvix.Library hiding (Type)
+import qualified Juvix.Library.HashMap as Map
+import LLVM.AST
+
+type SymbolTable = Map.T Symbol Operand
+
+type TypeTable = Map.T Symbol Type
+
+data SumInfo = S
+  { sum' :: Symbol,
+    offset :: Int,
+    tagSize' :: Word32
+  }
+  deriving (Show, Eq)
+
+-- | a mapping between the variant and the sum type along with
+-- the tag associated with it
+type VariantToType = Map.T Symbol SumInfo
```

Comment:
Why is this necessary for the custom monad?

Reply:
It's left in the shared file, we probably can move it to `1051` though.